plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/edit.tsx

```diff
@@ -79,20 +61,24 @@ const AddToCartOptionsEdit = ( props: BlockEditProps< Attributes > ) => {
 		unregisterListener,
 	] );
 
+	const innerBlocksTemplate = getInnerBlocksTemplate();
+
 	return (
 		<>
+			<InspectorControls>
+				<DowngradeNotice blockClientId={ props?.clientId } />
+			</InspectorControls>
```

Comment:
In the current shape, if you add Add to Cart with Options (Experimental) block, you'd see the notice even though you didn't upgrade. In scope of Product Collection I used `convertedFromProducts` attribute to mark blocks upgraded from Products (Beta) and showed the notice only for them.

https://github.com/woocommerce/woocommerce/blob/c7648abe5b5e1387c9e8ead887f1913b75e5443a/plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/index.tsx#L253

![add to cart](https://github.com/user-attachments/assets/4ac5747f-8dac-41e4-93ff-5cd4569688c8)

That's probably increases the scope of the whole feature but also we added couple of criteria to stop showing "Switch back" notice:

> // Upgrade Notice should be displayed only if:
> // - block is converted from Products
> // - user haven't acknowledged seeing the notice
> // - less than X hours since the notice was first displayed
> // - notice was displayed less than X times

https://github.com/woocommerce/woocommerce/blob/c7648abe5b5e1387c9e8ead887f1913b75e5443a/plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/index.tsx#L244C6-L262C3.

But considering the block is experimental I'll let you decide if that's blocking or not.

Reply:
Good points! I'm a bit on the fence on this. Personally, I would prefer to leave it as "always visible" for now, at least until we have a release plan and we have investigated extension support.

If we release this block in an experimental state, I think it will be better to have the button always visible to switch to the stable block. What do you think? We currently do something similar with the "Revert to Classic Template" button in templates, it's always visible no matter if the user actively switched to the blockified template or not.

Reply:
Yeah, I don't mind and it sounds ok! But wanted to mention this aspect anyway knowing we faced that earlier 🙌